.github/workflows/tests-on-packages.yml

```diff
@@ -0,0 +1,65 @@
+name: Run tests on APT packages
+
+on:
+  workflow_call:
+    inputs:
+      pkg_type:
+        required: true
+        type: string
+      revision_name:
+        required: true
+        type: string
+
+jobs:
+  TestPackagesWithLight:
+    strategy:
+      matrix:
+        distro:
+          - "debian:stretch"
+          - "debian:buster"
+          - "debian:bullseye"
+          - "debian:testing"
+          - "debian:sid"
+          - "ubuntu:xenial"
+          - "ubuntu:bionic"
+          - "ubuntu:focal"
+          - "ubuntu:impish"
+      fail-fast: false
+    runs-on: ubuntu-latest
+    container: ${{ matrix.distro }}
+    steps:
+      - name: Add ${{ inputs.pkg_type }} OSE repository and install syslog-ng (for Debian based OS)
+        run: |
+          apt-get update -qq
+          apt-get install -y wget gnupg2 ca-certificates apt-transport-https
+          wget -qO - https://ose-repo.syslog-ng.com/apt/syslog-ng-ose-pub.asc | apt-key add -
```

Comment:
`apt-key` is deprecated since Ubuntu 20.10 ( and in Debian Bullseye too I guess) and will be removed:

https://askubuntu.com/questions/1328806/how-to-solve-apt-key-deprecated

This is how we add the OSE APT repo in the syslog-ng` docker image:
```
wget -qO - https://ose-repo.syslog-ng.com/apt/syslog-ng-ose-pub.asc | gpg --dearmor > /usr/share/keyrings/ose-repo-archive-keyring.gpg && \    
  echo "deb [signed-by=/usr/share/keyrings/ose-repo-archive-keyring.gpg] https://ose-repo.syslog-ng.com/apt/ stable debian-testing" | tee --append /etc/apt/sources.list.d/syslog-ng-ose.list    
```

I don't know if this is supported on older Ubuntus and Debians as well.
We should probably change the `README.md` file about the APT repo too.

Reply:
Thanks for the comment, I have  fixed this part.